MODULE.bazel

```diff
@@ -36,6 +35,11 @@ bazel_dep(name = "with_cfg.bzl", version = "0.6.0")
 bazel_dep(name = "abseil-cpp", version = "20240722.0.bcr.2")
 bazel_dep(name = "rules_shell", version = "0.3.0")
 
+# Depend on apple_support first and then rules_cc so that the Xcode toolchain
```

Comment:
This still looks pretty brittle. Can we explicitly register the toolchain in MODULE.bazel so that the order doesn't matter?

Reply:
```
apple_cc_configure = use_extension("@apple_support//crosstool:setup.bzl", "apple_cc_configure_extension")
use_repo(apple_cc_configure, "local_config_apple_cc", "local_config_apple_cc_toolchains")

register_toolchains("@local_config_apple_cc_toolchains//:all")
```

Reply:
That feels more brittle to me: we would be reaching into apple_support internals and we may end up registering more or less than apple_support itself would.

Depending on order isn't that bad since `bazel_dep`s can't be accidentally duplicated (that would be an error). But I do agree that we should probably have a better story around this.

@brentleyjones who I discussed this with before

Reply:
OK, that makes sense.